Source: Richard Brown
Upstream: Not yet
Reason: --destdir got broken upstream
diff --git a/setup.rb b/setup.rb
index e819771..a66b6cd 100644
--- a/setup.rb
+++ b/setup.rb
@@ -95,11 +95,11 @@ require 'rbconfig'
 require 'tmpdir'
 require 'pathname'
 
+
 unless install_destdir.empty? then
   default_dir = Pathname.new(Gem.default_dir)
-  top_dir = Pathname.new(RbConfig::TOPDIR)
   ENV['GEM_HOME'] ||= File.join(install_destdir,
-                                default_dir.relative_path_from(top_dir))
+                                default_dir)
 end
 
 include FileUtils::Verbose
@@ -137,12 +137,8 @@ else
 end
 
 unless install_destdir.empty?
-  top_dir = Pathname.new(RbConfig::TOPDIR)
-  lib_dir_p = Pathname.new(lib_dir)
-  bin_dir_p = Pathname.new(bin_dir)
-
-  lib_dir = File.join install_destdir, lib_dir_p.relative_path_from(top_dir)
-  bin_dir = File.join install_destdir, bin_dir_p.relative_path_from(top_dir)
+  lib_dir = File.join install_destdir, lib_dir
+  bin_dir = File.join install_destdir, bin_dir
 end
 
 mkdir_p lib_dir
